annotations-runtime/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScript.java

```diff
@@ -148,14 +161,65 @@ protected Object invokeUriHandlerMethod(final AnnotationWebScriptRequest request
 	protected void handleUriMethodReturnValue(HandlerMethods handlerMethods, final AnnotationWebScriptRequest request,
                                               final AnnotationWebscriptResponse response, final Object returnValue) throws Exception {
         Resolution resolution = null;
-        if (returnValue instanceof Map) {
+        if (returnValue instanceof Map) { // returning a map will result in response template by default.
             resolution = new TemplateResolution((Map<String, Object>) returnValue);
-        } else if (returnValue instanceof String) {
+        } else if (returnValue instanceof String) { // returning a string will also result in a template response.
             resolution = new TemplateResolution((String)returnValue);
         } else if (returnValue instanceof Resolution) {
             resolution = (Resolution) returnValue;
         }
-        if (this.handlerMethods.useResponseTemplate()) {
+
+        /**
+         * If the method is annotated with {@link org.springframework.web.bind.annotation.ResponseBody}
+         * The response should be serialized automatically by what the request asked in the accept header.
+         * If the accept header is not present, the default format value of the method is used.
+         * If there is no default format available, an exception is thrown.
+         */
+        if (handlerMethods.useResponseBody()){
+            String[] responseTypes = request.getHeaderValues("Accept");
+
+            MediaType acceptResponseType = MediaType.ALL; // default
+
+            if (responseTypes != null){
+                acceptResponseType = MediaType.parseMediaType(responseTypes[0]);
+            } else {
+                String defaultResponse = this.getDescription().getDefaultFormat();
+                if (defaultResponse != null && !defaultResponse.isEmpty()) {
+                    acceptResponseType = MediaType.parseMediaType(defaultResponse);
+                }
+            }
+
+            HttpMessageConverter converter = null;
+
+            for(HttpMessageConverter messageConverter : this.messageConverters){
+                if (messageConverter.canWrite(returnValue.getClass(), acceptResponseType)){
+                    converter = messageConverter;
+                    break;
+                }
+            }
+
+            if(converter == null) {
+                // unsupported type requested
+                // should throw error.
+
+                List<MediaType> supported = new ArrayList<MediaType>();
+
+                for(HttpMessageConverter messageConverter : this.messageConverters){
+                    supported.addAll(messageConverter.getSupportedMediaTypes());
+                }
+
```

Comment:
Add in comments here that this is also thrown if `@RoolElem` is missing on xml-models